src/main/webapp/js/instructorFeedbackResultsAjaxByGQR.js

```diff
@@ -28,35 +24,40 @@ $(document).ready(function() {
                 displayAjaxRetryMessageForPanelHeading(displayIcon);
             },
             success: function(data) {
-                if (numPanels === 0) {
-                    numPanels += $('.panel-collapse').length;
-                }
+                var $sectionBody = $(panelBody[0]);
 
                 if (typeof data === 'undefined') {
-                    $(panelBody[0]).html('The results is too large to be viewed. Please choose to view '
-                                         + 'the results by questions or download the results.');
+                    $sectionBody.html('The results is too large to be viewed. '
+                            + 'Please choose to view the results by questions or download the results.');
                 } else {
-                    var appendedSection = $(data).find('#sectionBody-0').html();
+                    var $appendedSection = $(data).find('#sectionBody-0');
+                    var sectionId = $(panelHeading).attr('id').match(/section-(\d+)/)[1];
+
+                    if (removeEmptySection(sectionId, $appendedSection)) {
```

Comment:
`removeEmptySection` is doing too much, let's split it up into `isEmptySection` and `removeSection`.